### PR TITLE
sys-cluster/charliecloud: add 0.34

### PR DESCRIPTION
**Pull request by olifre**
This also drops the oldest release from tree and enables `py3.12` for the newly added release and the live ebuild.

### CODE REVIEW

File: sys-cluster/charliecloud/charliecloud-0.34.ebuild [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=8
 
-PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_COMPAT=( python3_{9..12} )
```
Comment (arthurzam) [inline sys-cluster/charliecloud/charliecloud-0.34.ebuild:6]:
```suggestion
PYTHON_COMPAT=( python3_{10..12} )
```

File: sys-cluster/charliecloud/charliecloud-9999.ebuild [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=8
 
-PYTHON_COMPAT=( python3_{9..11} )
+PYTHON_COMPAT=( python3_{9..12} )
```
Comment (arthurzam) [inline sys-cluster/charliecloud/charliecloud-9999.ebuild:6]:
```suggestion
PYTHON_COMPAT=( python3_{10..12} )
```

File: sys-cluster/charliecloud/charliecloud-0.34.ebuild [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	)"
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	local econf_args=()
+	econf_args+=(
```
Comment (arthurzam) [inline sys-cluster/charliecloud/charliecloud-0.34.ebuild]:
```
	local econf_args=(
```

**Review by arthurzam — CHANGES_REQUESTED** [review]

File: sys-cluster/charliecloud/charliecloud-0.34.ebuild (hunk 1 above)
Comment (olifre) [reply sys-cluster/charliecloud/charliecloud-0.34.ebuild]: Thanks, applied! 
Is there a rule on when to drop old langs?

File: sys-cluster/charliecloud/charliecloud-9999.ebuild (hunk 2 above)
Comment (olifre) [reply sys-cluster/charliecloud/charliecloud-9999.ebuild]: Thanks, applied.

File: sys-cluster/charliecloud/charliecloud-0.34.ebuild (hunk 3 above)
Comment (olifre) [reply sys-cluster/charliecloud/charliecloud-0.34.ebuild]: Good catch, applied, also to the `9999` ebuild.

File: sys-cluster/charliecloud/charliecloud-0.34.ebuild (hunk 1 above)
Comment (arthurzam) [reply sys-cluster/charliecloud/charliecloud-0.34.ebuild]: You can drop old targets when they are removed. For example we have removed python3.9 support globally (you can see this is now a NOP), so you can drop it. Another way to see it is by running `pkgcheck scan --commits`, which would flag all packages in the current patch set that retain the old COMPAT value.

**Review by arthurzam — APPROVED** [review]

File: sys-cluster/charliecloud/charliecloud-0.34.ebuild (hunk 1 above)
Comment (olifre) [reply sys-cluster/charliecloud/charliecloud-0.34.ebuild]: Thanks, that's very helpful! :+1:

**Pull request closed without merging**